File: Modules/HumanResource/EmployeeGroup/EmployeeGroup/EmployeeGroupForm.cs
using HRMSoftware.EmployeeGroup;
using HRMSoftware.Shift;
using Serenity.ComponentModel;
using System;
using System.Collections.Generic;

namespace HRMSoftware.EmployeeGroup.Forms;

[FormScript("EmployeeGroup.EmployeeGroup")]
[BasedOnRow(typeof(EmployeeGroupRow), CheckNames = true)]
public class EmployeeGroupForm
{
    public string Name { get; set; }
    [OneThirdWidth]
    public string ShiftColor { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    /*
    [OneThirdWidth]
    public DateTime StartDate { get; set; }
    [OneThirdWidth]
    public DateTime EndDate { get; set; }
    */
    [HalfWidth]
    public List<int> OccupationList { get; set; }
    [HalfWidth]
    public List<int> DepartmentList { get; set; }
    [HalfWidth]
    public List<int> DivisionList { get; set; }
    [HalfWidth]
    public List<int> JobGradeList { get; set; }
    [HalfWidth]
    public List<int> SectionList { get; set; }
    [HideOnInsert]
    public List<int> NewAddedEmployee { get; set; }
    public List<int> EmployeeList { get; set; }

    
    [IgnoreName, LabelWidth("0"),EmployeeGroupShiftEditor]
    public List<EmployeeGroupShiftPatternRow> Shifts { get; set; }

    [IgnoreName, LabelWidth("0"), EmployeeGroupShiftEditor]
    public List<EmployeeGroupShiftPatternRow> ActualShifts { get; set; }

}

File: Modules/ServerTypes/EmployeeGroup/EmployeeGroupForm.ts
import { StringEditor, TextAreaEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { EmployeeGroupShiftEditor } from "@/HumanResource/EmployeeGroup/EmployeeGroupShift/EmployeeGroupShiftEditor";
import { initFormType } from "@serenity-is/corelib/q";

export interface EmployeeGroupForm {
    Name: StringEditor;
    ShiftColor: StringEditor;
    Description: TextAreaEditor;
    OccupationList: LookupEditor;
    DepartmentList: LookupEditor;
    DivisionList: LookupEditor;
    JobGradeList: LookupEditor;
    SectionList: LookupEditor;
    EmployeeList: LookupEditor;
    NewAddedEmployee: LookupEditor;
    Shifts: EmployeeGroupShiftEditor;
    ActualShifts: EmployeeGroupShiftEditor;
}

export class EmployeeGroupForm extends PrefixedContext {
    static formKey = 'EmployeeGroup.EmployeeGroup';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeGroupForm.init)  {
            EmployeeGroupForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = LookupEditor;
            var w3 = EmployeeGroupShiftEditor;

            initFormType(EmployeeGroupForm, [
                'Name', w0,
                'ShiftColor', w0,
                'Description', w1,
                'OccupationList', w2,
                'DepartmentList', w2,
                'DivisionList', w2,
                'JobGradeList', w2,
                'SectionList', w2,
                'EmployeeList', w2,
                'NewAddedEmployee', w2,
                'Shifts', w3,
                'ActualShifts', w3
            ]);
        }
    }
}